aten/src/ATen/ExpandUtils.h

```diff
@@ -462,7 +462,8 @@ inline Tensor _sum_to(
     reduce_dims.push_back(i);
   }
   for (int64_t i = leading_dims; i < static_cast<int64_t>(sizes.size()); ++i) {
-    if (shape[i - leading_dims] == 1 && sizes[i] != 1) {
+    if (shape[i - leading_dims] == 1 &&
+        TORCH_GUARD_SIZE_OBLIVIOUS(sym_ne(sizes[i], 1))) {
```

Comment:
I'm starting to wonder whether we should have the size_oblivious = True by default...

Reply:
This is what we used to have, we introduced `guard_size_oblivious` because these tests were completely uncontrolled and we wanted more control over where they happen.

Reply:
so there are comparisons where we don't want to have size_oblivious = True?

Reply:
Yeah. In particular, comparisons in user code definitely should not have size oblivious on. We want the error there.